src/database/utilities/getIndicatorData.js

```diff
@@ -0,0 +1,138 @@
+import { UIDatabase } from '..';
+import { createRecord } from './createRecord';
+
+/**
+ * Returns a filter method for finding a subset of an array of IndicatorAttribute objects by period.
+ * @param {Array.<IndicatorAttribute>} filterPeriod
+ * @return {Func}
+ */
+const filterByPeriod = filterPeriod => ({ period }) => period.id === filterPeriod.id;
+
+/**
+ * Returns a filter method for getting the intersection of two arrays of IndicatorValue objects.
+ * @param {Array.<IndicatorValue>} values
+ * @return {Func}
+ */
+const filterByValues = values => value => includesValue(values, value);
+
+/**
+ * Returns a find method for querying an array of indicatorValue for the index of a given object.
+ * @param {IndicatorValue} findValue
+ * @return {Func}
+ */
+const findByValue = findValue => value => findValue.id === value.id;
+
+/**
+ * Query if an IndicatorValue array contains a given object.
+ * @param {Array.<IndicatorValue>} values
+ * @param {IndicatorValue} value
+ */
+const includesValue = (values, value) => values.some(findByValue(value));
+
+/**
+ * Create a new indicator value.
+ * @param {IndicatorAttribute} row
+ * @param {IndicatorAttribute} column
+ * @param {Period} period
+ * @return {IndicatorValue}
+ */
+const createIndicatorValue = (row, column, period) => {
+  let record;
+  UIDatabase.write(() => {
+    record = createRecord(UIDatabase, 'IndicatorValue', row, column, period);
+  });
+  return record;
+};
+
+/**
+ * Update indicator value object with new value.
+ * @param {IndicatorValue} indicatorValue
+ * @param {string} value
+ */
+const updateIndicatorValue = (indicatorValue, value) => {
+  UIDatabase.write(() => {
+    UIDatabase.update('IndicatorValue', { ...indicatorValue, value });
+  });
+};
+
+/**
+ * Get all indicator values for a given period.
+ * @param {*} period
+ */
+const getPeriodIndicatorValues = period =>
+  UIDatabase.objects('IndicatorValue').filter(filterByPeriod(period));
```

Comment:
`const getPeriodIndicatorValues = period =>
  UIDatabase.objects('IndicatorValue').filtered('period.id == $0', period.id);` is working fine for me 🤣

Reply:
Yup works for me too, and is a lot quicker. 7b9bf6f59c7a51869523e3067437ddbb58e8168f.